### PR TITLE
Add Etherpad Widget

### LINKED ISSUE #88: Add Etherpad widget
If possible, self-host. If using upstream (scalar), open a new ticket to add self-hosted option.

Related: https://github.com/turt2live/matrix-dimension/issues/38

### PR DESCRIPTION
**Pull request by MTRNord**
This PR adds support for Etherpad Widgets and closes #88 .
It propably is very simple to port this code for usage with #116 

## Todo:
- [x] Add missing Avatar
- [x] Rework `wrapUrl()` "hack"
- [x] Fix Names don't have the roomID in them
- [x] Fix Avatar Size and Position

Screenshots of changes to the Add Widget and Change Widget Menu:

#### Add Widget Default Server
grafik

### Add Widget Custom Server
grafik

### Edit Widget Default Server
grafik

### Edit Widget Custom Server
grafik

### CODE REVIEW

File: web/app/configs/widget/etherpad/etherpad-config.component.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        );
+    }
+
+    getPadURL(widget?: Widget): string {
+        let url: string;
+        if (widget) {
+            if (this.editUseCustomServer) {
+                url = widget.data.newPadServer + this.roomId + "_" + widget.data.newPadName;
+            } else {
+                url = "https://demo.riot.im/etherpad/p/" + this.roomId + "_" + widget.data.newPadName;
```
Comment (turt2live) [inline web/app/configs/widget/etherpad/etherpad-config.component.ts]: The default server should be a setting in the widget configuration file

**Comment by turt2live** [comment]
I've merged this manually to make the URL configurable

**Comment by MTRNord** [comment]
@turt2live sorry I just had no time to finish this :/

**Comment by turt2live** [comment]
No problem. Implementing it is greatly appreciated!

**Pull request merged**